modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/asset/entry/provider/test/AssetListAssetEntryProviderTest.java

```diff
@@ -73,6 +107,69 @@ public void setUp() throws Exception {
 			_group.getGroupId(), TestPropsValues.getUserId());
 	}
 
+	@Test
+	public void testCombineSegmentsOfDynamicCollection() throws Exception {
+		Dictionary<String, Object> dictionary =
+			HashMapDictionaryBuilder.<String, Object>put(
+				"enabled", true
+			).build();
+
+		try (ConfigurationTemporarySwapper configurationTemporarySwapper =
+				new ConfigurationTemporarySwapper(
+					"com.liferay.asset.list.web.internal.configuration." +
+						"FFCollectionsVariationsPrioritizationConfiguration",
+					dictionary)) {
+
+			AssetListEntry assetListEntry =
+				_assetListEntryLocalService.addAssetListEntry(
+					TestPropsValues.getUserId(), _group.getGroupId(),
+					"Dynamic title", AssetListEntryTypeConstants.TYPE_DYNAMIC,
+					null, _serviceContext);
+
+			SegmentsEntry segmentsEntry1 = _addSegmentsEntry(
+				_group.getGroupId(), TestPropsValues.getUser());
+
+			SegmentsEntry segmentsEntry2 = _addSegmentsEntry(
+				_group.getGroupId(), TestPropsValues.getUser());
+
+			long[] segmentsEntryIds = {
+				segmentsEntry1.getSegmentsEntryId(),
+				segmentsEntry2.getSegmentsEntryId()
+			};
+
+			AssetListEntrySegmentsEntryRel variation1 =
+				AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
+					_group.getGroupId(), assetListEntry,
+					segmentsEntry1.getSegmentsEntryId());
+
+			AssetListEntrySegmentsEntryRel variation2 =
+				AssetListTestUtil.addAssetListEntrySegmentsEntryRel(
```

Comment:
Following Liferay's style book, this variable should be called assetListEntrySegmentsEntryRel2